Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php

```diff
@@ -191,50 +213,79 @@ protected function buildModalHeading($label, $current_node) {
 	const PROP_POINTS = "points";
 	const PROP_STATUS = "status";
 
-	protected function buildForm() {
-		$form = new ilAsyncPropertyFormGUI();
+	protected function buildForm(\ilObjStudyProgramme $prg, string $submit_action) : ILIAS\UI\Component\Input\Container\Form\Standard {
+		$ff = $this->input_factory->field();
+		$tf = $this->trafo_factory;
+		$txt = function($id) { return $this->lng->txt($id); };
+		$sp_types = ilStudyProgrammeType::getAllTypesArray();
+		$status_options = self::getStatusOptions();
+		return $this->input_factory->container()->form()->standard(
+			$submit_action,
+			[
+				$ff->section(
+					[
+						self::PROP_TITLE =>
+							$ff->text($txt("title"))
+								->withValue($prg->getTitle())
+								->withRequired(true),
+						self::PROP_DESC =>
+							$ff->text($txt("description")) // TODO: change this to textarea once available
+								->withValue($prg->getDescription())
+					],
+					$txt("prg_edit"),
+					""
```

Comment:
One thing I do not like is, that the title and description come after the content in the section.